README.md

```diff
@@ -184,6 +187,89 @@ const purchaseOrder = poResponse.readResponse.record as PurchaseOrder;
 While TypeSuite currently supports types that can be written back to NetSuite, we don't yet have examples. Stay tuned
 for more!
 
+### A note on instantiated objects vs object literals
+
+Building requests to the NetSuite SuiteTalk API frequently involves instantiating quite a few objects.  In order to
+minimize the amount of code required to build all the necessary objects, in many cases an object literal can be
+provided instead of a fully instantiated object.  There are two cases where a fully instantiated object is
+required.  The first case is the outermost object, typically the Request object, which must be created by calling
+the class' constructor. In the example below, a GetRequest is explicitly instantiated via `new`.
+
+```ts
+const getRequest = new GetRequest({
+  ...
+  });
+```
+
+The second case where a fully instantiated object is required is where you provide an object that is a subclass of
+type that is expected for that field.  For example, GetRequest has single field called `baseRef` whose type is
+`PlatformCore.BaseRef`.  `PlatformCore.BaseRef` is an abstract type that isn't meant to be used directly.  Instead,
+you provide one of its subclasses, which includes `RecordRef`.  In order for the correct xml for this request to
+be generated, you must provide a fully instantiated `RecordRef`. 
+
+```ts
+const getRequest = new GetRequest({
+    baseRef: new RecordRef({
+      ...
+    }),
+  });
+```
+
+In all other cases you can simply supply an object literal with the data needed for your request.  Below is an
+example of a `SearchRequest` that uses a `TransactionSearchBasic`.  The `status` and `type` fields of
+`TransactionSearchBasic` have a type of `PlatformCore.SearchEnumMultiSelectField`, but it is not necessary to create
+these objects by calling `new`.
+
+```ts
+const searchRequest = new SearchRequest({
+  searchRecord: new TransactionSearchBasic({
+    status: {
+      operator: "anyOf",
+      searchValue: ["_purchaseOrderPendingReceipt"],
+    },
+    type: {
+      operator: "anyOf",
+      searchValue: ["_purchaseOrder"],
+    },
+  }),
+});
+```
+
+## Current limitations
+
+### API Version
+
+As mentioned above, TypeSuite only supports the [2019_2 WSDL](https://webservices.netsuite.com/wsdl/v2019_2_0/netsuite.wsdl).
+
+### Available ports
+
+TODO: Quick implementation or write up
+
+### Date handling
+
+Right now, you must convert Dates to a string manually.  This is due to a complication with one of
+the underlying libraries that TypeSuite uses and will be fixed in a later release.  However, it will always be
+possible to provide a string for any field in the NetSuite API that holds a Date.
```

Comment:
Same.

Would be good to have an example of what the date format is (`YYYY-MM-DDThh:mm:ss.sssZ`?) as well

Reply:
Formatting change done.

Reply:
As for an example of the date format, there's already an example of creating and using a DateFormatter in our example code.  Plus, any formatting requirements come from the WSDL, not from anything we are doing.